### ISSUE TITLE
platforms/chip: quickstart sample/doc broken

### ISSUE THREAD

**Issue by ahmetb** [hybridgroup/gobot#623]
I'm following https://gobot.io/documentation/platforms/chip/ on macOS and seeing a compilation error in the sample program:

    undefined: gobot.On

## Repro steps

I ran `go get -d -u gobot.io/x/gobot/...` (worked)

Pasted the sample listed to main.go:

```
package main

import (
    "fmt"

    "gobot.io/x/gobot"
    "gobot.io/x/gobot/drivers/gpio"
    "gobot.io/x/gobot/platforms/chip"
)

func main() {
    chipAdaptor := chip.NewAdaptor()
    button := gpio.NewButtonDriver(chipAdaptor, "XIO-P0")

    work := func() {
        gobot.On(button.Event("push"), func(data interface{}) {
            fmt.Println("button pressed")
        })

        gobot.On(button.Event("release"), func(data interface{}) {
            fmt.Println("button released")
        })
    }

    robot := gobot.NewRobot("buttonBot",
        []gobot.Connection{chipAdaptor},
        []gobot.Device{button},
        work,
    )

    robot.Start()
}
```

then I ran `GOARM=7 GOARCH=arm GOOS=linux go build .` as instructed and this cmd gives error:

    ./main.go:16:9: undefined: gobot.On
    ./main.go:20:9: undefined: gobot.On

**Comment by ahmetb** [comment]
similarly chip-pro instructions are broken, too https://gobot.io/documentation/platforms/chip-pro/

It seems like `gobot.On` isn't really a thing anymore. But I can't really find an `On` method for `platforms/chip` package to handle button press to refactor/fix the example code.

**Comment by Skgamer38** [comment]
I have an arduino nano with a atmel atmega328p microcontroller and idk if it can emulate a HID, to act like a keyboard and mouse. Someone can help me?

**Comment by gen2thomas** [comment]
Also there is a mismatch in dragonboard docu and example. I will fix this together.

**Comment by gen2thomas** [comment]
available since v2.2.0
